keras_cv/layers/object_detection/anchor_generator.py

```diff
@@ -0,0 +1,272 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow import keras
+
+from keras_cv import bounding_box
+
+
+class AnchorGenerator(keras.layers.Layer):
+    """AnchorGenerator generates anchors for multiple feature maps.
+
+    AnchorGenerator takes multiple scales and generates anchor boxes based on the anchor
+    sizes, scales, aspect ratios, and strides provided.  To invoke AnchorGenerator, call
+    it on the image that needs anchor boxes.
+
+    `anchor_sizes` and `strides` must match structurally - they are pairs.  Scales and a
```

Comment:
Let's rename `anchor_sizes` to be `sizes`

Reply:
Sounds good to me.